sdk/js/__tests__/Client.spec.js

```diff
@@ -242,17 +242,30 @@ describe('DVCClient tests', () => {
                 return Promise.resolve(testConfig)
             })
             saveEntity_mock.mockResolvedValue({})
-            const client = new DVCClient('test_env_key', { user_id: 'user1' }, {enableEdgeDB: true})
-            await client.onClientInitialized()
-            getConfigJson_mock.mockClear()
-            saveEntity_mock.mockClear()
+            const client = new DVCClient('test_env_key', { user_id: 'user1' }, { enableEdgeDB: true })
             await client.identifyUser(newUser)
 
             expect(getConfigJson).toBeCalledWith('test_env_key', expect.objectContaining({ user_id: 'user2' }), true)
             expect(saveEntity_mock).toBeCalledWith(expect.objectContaining(newUser), 'test_env_key')
             expect(saveEntity_mock).toBeCalledWith(expect.any(DVCPopulatedUser), 'test_env_key')
         })
 
+        it('should not send a request to edgedb after getting the config for an anonymous user', async () => {
+            const newUser = { isAnonymous: true, country: 'CA' }
+            getConfigJson_mock.mockImplementation(() => {
+                return Promise.resolve(testConfig)
+            })
+
+            const client = new DVCClient('test_env_key', 
+                new DVCPopulatedUser({ isAnonymous: true }, { enableEdgeDB: true }), 
+                { enableEdgeDB: true }
+            )
+            await client.identifyUser(newUser)
+
+            expect(getConfigJson).toBeCalled()
+            expect(saveEntity_mock).not.toBeCalled()
```

Comment:
a good way to make sure your test is working properly is to run it against the unchanged SDK code, and make sure it fails. Then add your fix and make sure it now passes. I have a feeling this test will always pass.